tests/queries/0_stateless/03394_pr_insert_select_threads.sql

```diff
@@ -0,0 +1,35 @@
+-- Tags: long, no-parallel
+
+SET enable_analyzer=1; -- parallel distributed insert select for replicated tables works only with analyzer
+SET parallel_distributed_insert_select=2;
+
+DROP TABLE IF EXISTS t_mt_source;
+DROP TABLE IF EXISTS t_mt_target;
+
+CREATE TABLE t_mt_source (k UInt64, v String) ENGINE = MergeTree() ORDER BY k;
+CREATE TABLE t_mt_target (k UInt64, v String) ENGINE = MergeTree() ORDER BY ();
+
+SYSTEM STOP MERGES t_mt_source;
+INSERT INTO t_mt_source SELECT number as k, toString(number) as v FROM system.numbers LIMIT 1e6 SETTINGS max_block_size=1000, min_insert_block_size_rows=1000;
+SELECT count() FROM system.parts WHERE database = currentDatabase() and table = 't_mt_source';
+
+SET cluster_for_parallel_replicas='test_cluster_one_shard_three_replicas_localhost', max_parallel_replicas=3, parallel_replicas_for_non_replicated_merge_tree=1, parallel_replicas_mark_segment_size=128;
+
+-- reduce block size to ensure that all threads will be used
+INSERT INTO t_mt_target SELECT * FROM t_mt_source SETTINGS allow_experimental_parallel_reading_from_replicas=1, max_threads=8, max_insert_threads=4, max_block_size=1000, min_insert_block_size_rows=1000, parallel_replicas_local_plan=0;
+INSERT INTO t_mt_target SELECT * FROM t_mt_source SETTINGS allow_experimental_parallel_reading_from_replicas=1, max_threads=8, max_insert_threads=4, max_block_size=1000, min_insert_block_size_rows=1000, parallel_replicas_local_plan=1, parallel_replicas_insert_select_local_pipeline=1;
+
+SYSTEM FLUSH LOGS query_log;
+SELECT
+    if(is_initial_query, 'inital', 'secondary'),
+    if(toUInt64OrZero(Settings['parallel_replicas_insert_select_local_pipeline']) == 0 and is_initial_query, 1, 8) threads_limit,
```

Comment:
The setting `parallel_replicas_insert_select_local_pipeline` is true by default, and, afais, we don't change it in the test. Why do we check it here?

Reply:
For job with PR on CI